invenio_indexer/api.py

```diff
@@ -187,6 +187,23 @@ def index(self, record, arguments=None, **kwargs):
             **arguments,
         )
 
+    def index_to_datastream(self, record, arguments=None, **kwargs):
+        """Index a record to a datastream index.
+
+        :param record: Record instance.
+        """
+        index = self.record_to_index(record)
+        arguments = arguments or {}
+        body = self._prepare_record(record, index, arguments, **kwargs)
+        index = self._prepare_index(index)
+
+        # We don't pass the id and version fields as it is a datastream index
```

Comment:
```suggestion
        # We don't pass the id and version fields as it is a datastream index - it won't accept id
```

Reply:
minor: couldn't we use the `self.client.create(...)` to basically do a `PUT /<datastream>/_create/<id>` request and avoid the whole `id` skipping? Then we would also rename this method to `create`.

Reply:
I decided not to use `id` and `create` because [it would be faster to index without id](https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-indexing-speed.html#_use_auto_generated_ids).

Reply:
@sakshamarora1 can you double-check this with @slint, and then we merge/release (minor release) and keep things moving forward?

Reply:
Since we're usually inserting just a couple of documents at a time I don't think we would gain much more from having the auto-generated IDs from ElasticSearch vs. being able to correlate DB entries with their equivalent indexed entry in OpenSearch.

Right now we run into the danger of e.g. accidentally indexing something and not being able to track it back in case we want to delete it.